onnxruntime/core/providers/shared/initializer_view/initializer_view.h

```diff
@@ -0,0 +1,55 @@
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#pragma once
+
+#include <cstdint>
+#include <optional>
+
+#include "core/framework/tensorprotoutils.h"
+#include "core/framework/tensor_shape.h"
+
+#include "gsl/gsl-lite.hpp"
+
+namespace onnxruntime {
+
+class InitializerView {
+ private:
+  InitializerView(const InitializerView&) = delete;
+  InitializerView& operator=(const InitializerView&) = delete;
+
+ public:
+  InitializerView() = default;
+
+  static common::Status Create(
+      const ONNX_NAMESPACE::TensorProto& tensor_proto, std::optional<InitializerView>& initializer);
+
+  const TensorShape& Shape() const {
+    return shape_;
+  }
+
+  int32_t Type() {
+    return data_type_;
+  }
+
+  template <class T>
+  gsl::span<const T> DataAsSpan() const {
+    // make sure data bytes are more than sizeof (T)";
```

Comment:
I think we can make this safer by enforcing that T matches data_type_. if we also need to access bytes (e.g. for memcpy), consider adding an additional member function like `gsl::span<const uint8_t> DataAsByteSpan() const`

Reply:
Make sence, Aligning with Tensor is good. We also offer a light interface if DataAsByteSpan is needed.